komfydb/execution/predicate.h

```diff
@@ -0,0 +1,51 @@
+#ifndef __PREDICATE_H__
+#define __PREDICATE_H__
+
+#include "komfydb/common/field.h"
+#include "komfydb/common/int_field.h"
+#include "komfydb/common/string_field.h"
+#include "komfydb/execution/op.h"
+#include "komfydb/storage/record.h"
+
+#include "absl/status/status.h"
+#include "absl/status/statusor.h"
+#include "komfydb/utils/status_macros.h"
+
+namespace {
+
+using namespace komfydb::common;
+using namespace komfydb::storage;
+
+}  // namespace
+
+class Predicate {
+ public:
+  enum Type {
+    COL_COL,
+    COL_CONST,
+  };
+
+  Predicate(Op op, int field, std::unique_ptr<Field> const_field);
+
+  Predicate(Op op, int field1, int field2);
```

Comment:
I'd declare this as `Predicate(int l_field, Op op, int r_field)`, as we do it like this in other places.